mmdet/core/visualization/image.py

```diff
@@ -27,17 +35,184 @@ def color_val_matplotlib(color):
     return tuple(color)
 
 
+def _get_adaptive_scales(areas, min_area=800, max_area=30000):
+    """Get adaptive scales according to areas.
+
+    The scale range is [0.5, 1.0]. When the area is less than
+    ``'min_area'``, the scale is 0.5 while the area is larger than
+    ``'max_area'``, the scale is 1.0.
+    Args:
+        areas (ndarray): The areas of bboxes or masks with the
+            shape of (n, ).
+        min_area: Lower bound areas for adaptive scales.
+            Default: 800.
+        max_area: Upper bound areas for adaptive scales.
+            Default: 30000.
+
+    Returns:
+        ndarray : The adaotive scales with the shape of (n, ).
+    """
+    scales = 0.5 + (areas - min_area) / (max_area - min_area)
```

Comment:
Here, the range of scales is [0.5, 1.5]. This may differ from the definition of [0.5, 1.0].